cppcache/integration/test/Position.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifndef POSITION_H
+#define POSITION_H
+
+/*
+ * @brief User class for testing the put functionality for object.
+ */
+
+#include <string>
+
+#include <geode/CacheableString.hpp>
+#include <geode/DataSerializable.hpp>
+
+namespace DataSerializableTest {
+
+using apache::geode::client::CacheableString;
+using apache::geode::client::DataInput;
+using apache::geode::client::DataOutput;
+using apache::geode::client::DataSerializable;
+
+class Position : public DataSerializable {
+ private:
+  int64_t avg20DaysVol;
+  std::string bondRating;
+  double convRatio;
+  std::string country;
+  double valueGain;
+  int64_t industry;
+  int64_t issuer;
+  double mktValue;
+  double qty;
+  std::string secId;
+  std::string secLinks;
+  std::wstring secType;
```

Comment:
We shouldn't be continuing the use of the platform ambiguous `std::wstring`. Instead we should use `std::u16string` or just `std::string` with utf8 encoding. In either case the stream just needs to serialize the correct string type for the the Java side to deserialize.

Reply:
Agreed. Just ratholed on this issue for 30 minutes. Not sure this test is the right place to be testing string types. I've changed this variable to just be std::string, and serializing with readString/writeString. The corresponding Position.java code uses String and readUTF/writeUTF for all of the string variables. Doesn't break any tests.

Reply:
You should continue to use `writeUTF` even with `std::string` if the Java side uses `writeUTF`. This method has a very specific way of encoding strings that is non-standard. While it won't affect this test it could encourage us towards doing the wrong things in the future or worse encouraging a consumer to.